src/main/php/lang/ast/Emitter.class.php

```diff
@@ -113,7 +113,7 @@ protected function emit() {
    * @return void
    */
   protected function emitOperator($result, $operator) {
-    throw new IllegalStateException('Unexpected operator '.$operator->value.' at line '.$operator->line);
+    throw new IllegalStateException("Unexpected operator {$operator->value} at line {$operator->line}");
```

Comment:
The latter is definitely more readable, we might want to keep this even if we decide not to merge this entire pull request.